patches/post/20231023-Redact-auth-token-in-debug-output.patch

```diff
@@ -0,0 +1,67 @@
+diff --git a/metal/v1/client.go b/metal/v1/client.go
+index 082bc38..46b7665 100644
+--- a/metal/v1/client.go
++++ b/metal/v1/client.go
+@@ -355,6 +355,9 @@ func parameterToJson(obj interface{}) (string, error) {
+ 
+ // callAPI do the request.
+ func (c *APIClient) callAPI(request *http.Request) (*http.Response, error) {
++    if len(request.Header.Get("X-Auth-Token")) != 0 {
++        request.Header.Set("X-Auth-Token", "**REDACTED**")
++    }
+ 	if c.cfg.Debug {
+ 		dump, err := httputil.DumpRequestOut(request, true)
+ 		if err != nil {
+@@ -370,6 +373,8 @@ func (c *APIClient) callAPI(request *http.Request) (*http.Response, error) {
+ 
+ 	dumpDeprecation(resp)
+ 
++    dumpDeprecation(resp)
++
+ 	if c.cfg.Debug {
+ 		dump, err := httputil.DumpResponse(resp, true)
+ 		if err != nil {
+@@ -819,3 +824,43 @@ func dumpDeprecation(resp *http.Response) {
+ 		}
+ 	}
+ }
++
++// dumpDeprecation logs headers defined by
++// https://tools.ietf.org/html/rfc8594
++// copied from https://github.com/packngo
++// for backwards compatibility
++func dumpDeprecation(resp *http.Response) {
```

Comment:
This is already being added in https://github.com/equinix-labs/metal-go/blob/aad975b164631294f52ea787558e8d24a2505735/patches/post/20230626-log-sunset-deprecation-headers.patch#L9

@ctreatma's comment is the way to go: https://github.com/equinix-labs/metal-go/pull/167/files#r1368975973